### PR TITLE
Prepare for an update to the XNNPACK submodule

### PR DESCRIPTION
**Pull request by digantdesai**
- Make USE_XNNPACK a dependent option on cmake minimum version 3.12
- Print USE_XNNPACK under CMake options summary, and print the availability from collect_env.py
- Skip mobile_optimizer based tests when XNNPACK is not available
- Update cmake version for xenial-py3.7-gcc5.4 image to 3.12.4
   - This is required for the backwards compatibility test. The PyTorch op schema is XNNPACK dependent. See, aten/src/ATen/native/xnnpack/RegisterOpContextClass.cpp for example. The nightly version is assumed to have USE_XNNPACK=ON, so with this change we ensure that the test build can also have XNNPACK.

### CODE REVIEW

**Comment by digantdesai** [comment]
@malfet, @kimishpatel, @iseeyuan - Can you please take a look? Thanks!

**Review by malfet — APPROVED** [review]
Looks good to me, but please consider printing some sort of warning if condition is not met

Also, to prevent silent regressions in nightlies, please add check that validates that  XNNPACK is available in nightly/release builds to
https://github.com/pytorch/builder/blob/6010816fcfe4b70958da25cc8db9fd61ceb53581/check_binary.sh#L324

File: test/jit/test_optimize_for_mobile_preserve_debug_info.py [hunk 1]
```diff
@@ -36,6 +37,9 @@ def check_replacement(
                     original_source_ranges[replacements[node.kind()]],
                 )
 
+    @unittest.skipUnless(torch.backends.xnnpack.enabled,
+                         " XNNPACK must be enabled for these tests."
+                         " Please build with USE_XNNPACK=1.")
```
[the remark this answers is not in the document]
Comment (kimishpatel) [reply test/jit/test_optimize_for_mobile_preserve_debug_info.py:42]: Good suggestion.

**Review by kimishpatel — APPROVED** [review]

**Comment by digantdesai** [comment]
> Looks good to me, but please consider printing some sort of warning if condition is not met

Done!

> Also, to prevent silent regressions in nightlies, please add check that validates that XNNPACK is available in nightly/release builds to https://github.com/pytorch/builder/blob/6010816fcfe4b70958da25cc8db9fd61ceb53581/check_binary.sh#L324

Like this? https://github.com/pytorch/builder/pull/964

File: test/test_xnnpack_integration.py [hunk 2]
```diff
@@ -20,6 +20,7 @@
                      " XNNPACK must be enabled for these tests."
                      " Please build with USE_XNNPACK=1.")
 @unittest.skipIf(TEST_WITH_TSAN, "TSAN fails with XNNPACK. Does not seem to have a good reason for failures.")
+@unittest.skipIf(PYTORCH_TEST_WITH_ROCM, "HACK, not sure why these fail on ROCM.")
```
Comment (digantdesai) [inline test/test_xnnpack_integration.py:23]: @kimishpatel - is this OK for now?

File: test/test_xnnpack_integration.py [hunk 3]
```diff
@@ -934,6 +936,7 @@ def forward(self, x):
                      " XNNPACK must be enabled for these tests."
                      " Please build with USE_XNNPACK=1.")
 @unittest.skipIf(TEST_WITH_TSAN, "TSAN is not fork-safe since we're forking in a multi-threaded environment")
+@unittest.skipIf(TEST_WITH_ROCM, "HACK, not sure why these fail on ROCM.")
```
Comment (kimishpatel) [inline test/test_xnnpack_integration.py:939]: This is ok for now. Do file a follow up task with details on which build failed and what you tried to resolve.

**Review by kimishpatel — APPROVED** [review]

File: test/test_xnnpack_integration.py (hunk 3 above)
Comment (digantdesai) [reply test/test_xnnpack_integration.py]: Created T112777496.

**Pull request closed without merging**